pic32/cores/pic32/WInterrupts.c

```diff
@@ -263,12 +265,22 @@ void __attribute__((nomips16,at_vector(_EXTERNAL_0_VECTOR),interrupt(IPL4SRS)))
 void __attribute__((interrupt(),nomips16)) ExtInt0Handler(void)
 #endif
 {
-
+	VEC_ENT_CODE(EXT_INT0,INT0IF,PIN_INT0);
```

Comment:
That could just be a single digitalRead in the attachInterrupt routine.

You should also I suppose ensure that IFS is cleared in detachInterrupt. Shouldn't make any difference since the interrupt is disabled, but best to be sure.